build.rs

```diff
@@ -3,11 +3,24 @@ use std::env;
 use git2::Repository;
 
 fn main() {
-    let repo = Repository::open(".").expect("should open git repository");
-    let head = repo.head().expect("should get HEAD");
-    let branch = head.shorthand().expect("should get branch name");
-    let commit = head.peel_to_commit().expect("should get HEAD commit");
-    let rev = commit.id().to_string();
+    let branch = if let Ok(describe) = env::var("GIT_DESCRIBE") {
```

Comment:
This will fail if you build locally using Nix with overriding output attributes (as you did for the CI).

Reply:
The CI fail was more about the openssh, I think
https://github.com/KoviRobi/himalaya/actions/runs/11935131223/job/33265788614#step:5:1618
the others worked? So rebasing on top of your changes it should now work.

I have given the environment variables preference, that way you don't have to remove/move the `.git` dir/file.

It works both locally without overriding:
```console
$ nix build .
$ ./result/bin/himalaya --version
himalaya v1.0.0 on linux musl x86_64, git flake-a0485ff rev a0485ff8d18482499bde88e177ff8d989b0d2e5e
```
and with overriding
```
$ nix build --expr "(builtins.getFlake \"git+file://${PWD}?shallow=1&rev=$(git rev-parse HEAD)\").outputs.packages.x86_64-linux.default.overrideAttrs (old: { cargo_build_options = old.cargo_build_options or [] ++ [\"-F\" \"oauth2\"];  GIT_DESCRIBE = \"$(git describe --always HEAD)\"; })"
```

I have also managed to override it in a flake, including cargo features:
https://github.com/KoviRobi/nixos-config/commit/b63057268b5fa39b16b14d987ed86232c8599034#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R86-R100 though some of this is naersk weirdness I'm not sure I got correct.